### PR TITLE
Add readuntil benchmarks

### PR DESCRIPTION
**Pull request by omus**
See https://github.com/JuliaLang/julia/pull/20621

### CODE REVIEW

File: src/string/StringBenchmarks.jl [hunk 1]
```diff
@@ -29,4 +29,27 @@ g = addgroup!(SUITE, "searchindex")
...
+target = ("A" ^ 5000) * "Z"
+g["backtracking"] = @benchmarkable readuntil(seekstart($buffer), $target)
+
+buffer = IOBuffer(randstring('A':'X', 40000) * target)
+target = "Y" * ("Z" ^ 999)
+g["no backtracking"] = @benchmarkable readuntil(seekstart($buffer), $target)
+
+buffer = IOBuffer(("bar" ^ 20000) * "ians")
+target = ("bar" ^ 300) * "ian"
+g["barbarian backtrack"] = @benchmarkable readuntil(seekstart($buffer), $target)
```
Comment (ararslan) [inline src/string/StringBenchmarks.jl]: Good name

Comment (omus) [reply src/string/StringBenchmarks.jl]: In my quest for good backtracking words to use I also found:
* assassin
* cincinatti
* conconscious
* murmurously
* reprepare

Comment (ararslan) [reply src/string/StringBenchmarks.jl]: Wow, all good names.

File: src/string/StringBenchmarks.jl [hunk 2]
```diff
@@ -29,4 +29,27 @@ g = addgroup!(SUITE, "searchindex")
...
+g = addgroup!(SUITE, "readuntil")
+for len in (1, 2, 1000, 50000)
+    g["target length $len"] = @benchmarkable readuntil(seekstart($buffer), $("A" ^ len))
+end
+
+buffer = IOBuffer(("A" ^ 50000) * "B")
+target = ("A" ^ 5000) * "Z"
+g["backtracking"] = @benchmarkable readuntil(seekstart($buffer), $target)
+
+buffer = IOBuffer(randstring('A':'X', 40000) * target)
```
Comment (vtjnash) [inline src/string/StringBenchmarks.jl]: use `randstring(RandUtils.SEED, 'A':'X', 40000)` for stable random values

Comment (vtjnash) [reply src/string/StringBenchmarks.jl]: actually, look like it should be `String(rand(RandUtils.SEED, 'A':'X', 40000))`, so it works on v0.5 and v0.6 too?

**Pull request merged**